$OpenBSD: patch-tools_clang_lib_Driver_ToolChains_Gnu_cpp,v 1.1 2017/09/01 06:40:28 ajacoutot Exp $

Use the integrated assembler by default on Sparc and Mips64.

Index: tools/clang/lib/Driver/ToolChains/Gnu.cpp
--- tools/clang/lib/Driver/ToolChains/Gnu.cpp.orig
+++ tools/clang/lib/Driver/ToolChains/Gnu.cpp
@@ -2335,6 +2335,9 @@ bool Generic_GCC::IsIntegratedAssemblerDefault() const
   case llvm::Triple::systemz:
   case llvm::Triple::mips:
   case llvm::Triple::mipsel:
+  case llvm::Triple::sparc:
+  case llvm::Triple::sparcel:
+  case llvm::Triple::sparcv9:
     return true;
   case llvm::Triple::mips64:
   case llvm::Triple::mips64el:
@@ -2342,7 +2345,8 @@ bool Generic_GCC::IsIntegratedAssemblerDefault() const
     // identify the ABI in use (Debian) or only use N64 for MIPS64 (Android).
     // Other targets are unable to distinguish N32 from N64.
     if (getTriple().getEnvironment() == llvm::Triple::GNUABI64 ||
-        getTriple().isAndroid())
+        getTriple().isAndroid() ||
+        getTriple().isOSOpenBSD())
       return true;
     return false;
   default:
